### PR TITLE
add spfft

### PR DESCRIPTION
**Pull request by ltalirz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ltalirz** [comment]
@adhocman 

Currently the build fails (not sure whether you tried it locally before).
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=305597&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823

One of the first questions to answer is whether you want to provide the package on all three OSs (linux, osx and windows) or just on a subset. One can also start just with linux if that's easier, but at least osx would probably make sense as well.

As for fixing the build process, you can put your suggestions or you're also welcome to push to my fork directly (I've just invited you as a collaborator).

**Comment by ltalirz** [comment]
@conda-forge-admin , please rerender

**Comment by AdhocMan** [comment]
Thanks for creating the PR!
The list of maintainers looks good to me. I've added a few changes, which should have solved some of the issues.
In general, SpFFT, SPLA and SIRIUS only support Linux and macOS, so I've disabled the windows build.

**Comment by ltalirz** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by ltalirz** [comment]
@AdhocMan if you like, you can already go ahead and prepare a similar PR for [spla](https://github.com/eth-cscs/spla)

File: recipes/spfft/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mkdir -p build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX="${PREFIX}"  -DSPFFT_MPI=ON -DSPFFT_OMP=ON -DSPFFT_GPU_BACKEND=OFF -DSPFFT_GPU_DIRECT=OFF -DSPFFT_SINGLE_PRECISION=ON -DSPFFT_STATIC=OFF -DSPFFT_FORTRAN=ON -DSPFFT_BUILD_TESTS=OFF
+make -j2 install
```
Comment (ocefpaf) [inline recipes/spfft/build.sh:4]: Is there a `make check`, `ctest` or `make test` that we can run here?

Comment (AdhocMan) [reply recipes/spfft/build.sh:15]: There are tests that can be run, which I've added now.

File: recipes/spfft/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+mkdir -p build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX="${PREFIX}"  -DSPFFT_MPI=ON -DSPFFT_OMP=ON -DSPFFT_GPU_BACKEND=OFF -DSPFFT_GPU_DIRECT=OFF -DSPFFT_SINGLE_PRECISION=ON -DSPFFT_STATIC=OFF -DSPFFT_FORTRAN=ON -DSPFFT_BUILD_TESTS=ON
+make -j2
```
Comment (xhochy) [inline recipes/spfft/build.sh:4]:
```suggestion
make -j${CPU_COUNT}
```

File: recipes/spfft/build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+mkdir -p build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX="${PREFIX}"  -DSPFFT_MPI=ON -DSPFFT_OMP=ON -DSPFFT_GPU_BACKEND=OFF -DSPFFT_GPU_DIRECT=OFF -DSPFFT_SINGLE_PRECISION=ON -DSPFFT_STATIC=OFF -DSPFFT_FORTRAN=ON -DSPFFT_BUILD_TESTS=ON
```
Comment (xhochy) [inline recipes/spfft/build.sh:3]:
```suggestion
cmake .. ${CMAKE_ARGS}  -DSPFFT_MPI=ON -DSPFFT_OMP=ON -DSPFFT_GPU_BACKEND=OFF -DSPFFT_GPU_DIRECT=OFF -DSPFFT_SINGLE_PRECISION=ON -DSPFFT_STATIC=OFF -DSPFFT_FORTRAN=ON -DSPFFT_BUILD_TESTS=ON
```

File: recipes/spfft/build.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# workaround to run Open MPI in docker container
+export OMPI_MCA_plm_rsh_agent=sh
+
+# run tests
+./tests/run_local_tests
+mpirun -n 2 ./tests/run_mpi_tests
+
+# rebuild without tests to disable timers and make only API symbols visible
+cmake .. -DSPFFT_BUILD_TESTS=OFF
+make -j2 install
```
Comment (xhochy) [inline recipes/spfft/build.sh:15]:
```suggestion
make -j${CPU_COUNT} install
```

File: recipes/spfft/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/eth-cscs/SpFFT/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 541f6d92993655b68404cfea60d84ce0a7644eed49dff7e7437db818127700f9
+
+build:
+  number: 0
+  skip: True  # [ win ]
```
Comment (xhochy) [inline recipes/spfft/meta.yaml:14]:
```suggestion
  skip: True  # [win]
```

File: recipes/spfft/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    - {{ mpi }}
+    - fftw
+    - libgomp      # [linux]
+    - llvm-openmp  # [osx]
+  run:
+    - {{ mpi }}
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib64/libspfft${SHLIB_EXT}" || test -f "${PREFIX}/lib/libspfft${SHLIB_EXT}"
```
Comment (xhochy) [inline recipes/spfft/meta.yaml:32]: Libraries should always be in `lib` on conda-forge.
```suggestion
    - test -f "${PREFIX}/lib/libspfft${SHLIB_EXT}"
```

**Comment by AdhocMan** [comment]
Thanks for the review, the suggested changes are now included.

**Pull request merged**